src/plugins/data/public/ui/dataset_navigator/lib/catalog_cache/cache_loader.tsx

```diff
@@ -0,0 +1,470 @@
+/*
```

Comment:
@kavilla  @Swiddis  Why is Cache manager, Loader and intercepts copied here? This already exists today in the core: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source_management/framework/catalog_cache/cache_manager.ts

Reply:
Adding in the design we agreed on. 
![Screenshot 2024-07-23 at 12 11 09 PM](https://github.com/user-attachments/assets/6ed0b616-3d2a-4b73-9d2b-bb66a92afca0)

Reply:
This was done for speed and is temporary. The cache that was brought in needs rework since it depends on the observability dashboards plugin to work, which makes it a circular dependency and also makes calls to load and update the cache, not ideal. Also Ryans changes was only merged in yesterday.

Reply:
If this is temp and for speeding up things, I'm good. We can surely use the one in datasource managment plugin in core as it doesn't depend on Observability, once things stabilize post 2.16.

Reply:
+1